06.bowling_object/lib/game.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  STRIKE_MARK = 'X'
+
+  attr_reader :frames
+
+  def initialize(shot_marks)
+    delimited_marks = parse_marks(shot_marks.split(','))
+    @frames = delimited_marks.map { |m| Frame.new(m) }
+  end
+
+  def score
+    pin_score = @frames.map(&:score).sum
+    strike_bonus = calculate_strike_bonus
+    spare_bonus = calculate_spare_bonus
+    pin_score + strike_bonus + spare_bonus
+  end
+
+  private
+
+  def parse_marks(all_marks)
+    frames = []
+    9.times do
+      mark_pair = all_marks.shift(2)
+      frames << (mark_pair.include?(STRIKE_MARK) ? [mark_pair.first] : mark_pair)
+      all_marks.unshift(mark_pair.last) if mark_pair.include?(STRIKE_MARK)
+    end
+
+    frames << all_marks
```

Comment:
`<<`は要素を追加するのが目的で、戻り値が返るのは副作用的なものだと思うので、`<<`をメソッドの戻り値にするのはあまり適切ではないと考えます。
僕だったらこんなふうに書きますね。

```suggestion
    [*frames, all_marks]
```

Reply:
ご指摘ありがとうございます。修正しました。
最後に評価（実行）された式の結果が戻り値になる、とだけ認識していました。
戻り値として適切な式とそうではない式があるのですね...